File: ABC/ABC333/B/main.cc
#include <iostream>

int Dist(char a, char b) {
  int d = std::abs((a - 'A') - (b - 'A'));
  return std::min(d, 5 - d);
}

int main() {
  char a, b, c, d;
  std::cin >> a >> b >> c >> d;
  std::cout << (Dist(a, b) == Dist(c, d) ? "Yes" : "No" ) << std::endl;
  return 0;
}

File: ABC/ABC333/B/Main.java
import java.util.Scanner;

public class Main {
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    char[] s = sc.next().toCharArray();
    char[] t = sc.next().toCharArray();
    sc.close();
    System.out.println(dist(s) == dist(t) ? "Yes" : "No");
  }

  public static int dist(char[] s) {
    int d = Math.abs((s[0] - 'A') - (s[1] - 'A'));
    return Math.min(d, 5 - d);
  }
}